src/sbn_probability.cpp

```diff
@@ -264,24 +275,37 @@ EigenVectorXd SBNProbability::ExpectationMaximization(
       // Last line of the section on EM in doc/tex.
       score_history[em_idx] += m_tilde_for_positive_alpha.dot(sbn_parameters);
     }
+    // SHJ: Print the score in the below line.
+    // Uncommenting it but leaving it incase we need to do further debugging.
+    // Remove it at some point.
+    //std::cout << std::setprecision(20) << score_history[em_idx] << std::endl;
     // Return if we've converged according to score.
     if (em_idx > 0) {
       double scaled_score_improvement =
           (score_history[em_idx] - score_history[em_idx - 1]) /
           fabs(score_history[em_idx - 1]);
       // To monitor correctness of EM, we check to ensure that the score is
       // monotonically increasing (modulo numerical instability).
-      Assert(scaled_score_improvement > -EPS, "Score function decreased.");
+      // SHJ: -EPS is too big, I noticed the assertion failure for
+      // scaled_score_improvement of -6e-16. Using ERR_TOLERANCE
+      // which is 1e-10.
+      //Assert(scaled_score_improvement > -EPS, "Score function decreased.");
```

Comment:
That's fine, but let's take out the commented code.

Reply:
Removing commented Assert and 1e-10 from the comments.